### PR TITLE
chore: clean up lints

### PR DESCRIPTION
**Pull request by DaniPopes**
Too many `#[allow]`s for no good reason.
- Fix some TODOs
- reduce `unused` to individual lints (`unused` is a massive lint group and I generally dislike enabling this ever)

### CODE REVIEW

File: crates/net/downloaders/src/test_utils/mod.rs [hunk 1]
```diff
@@ -51,12 +50,12 @@ pub(crate) fn generate_bodies(
 /// Generate a set of bodies, write them to a temporary file, and return the file along with the
 /// bodies and corresponding block hashes
 pub(crate) async fn generate_bodies_file(
-    rng: RangeInclusive<u64>,
+    range: RangeInclusive<u64>,
 ) -> (tokio::fs::File, Vec<SealedHeader>, HashMap<B256, BlockBody>) {
-    let (headers, mut bodies) = generate_bodies(0..=19);
-    let raw_block_bodies = create_raw_bodies(headers.clone().iter(), &mut bodies.clone());
+    let (headers, mut bodies) = generate_bodies(range);
```
Comment (mattsse) [inline crates/net/downloaders/src/test_utils/mod.rs:55]: this change alters previous logic, 
I guess should be fine here

File: crates/payload/basic/src/lib.rs [hunk 2]
```diff
@@ -648,7 +648,7 @@ pub struct PayloadConfig {
     /// The chain spec.
     chain_spec: Arc<ChainSpec>,
     /// The rollup's compute pending block configuration option.
-    /// TODO(clabby): Implement this feature.
+    // TODO(clabby): Implement this feature.
```
Comment (mattsse) [inline crates/payload/basic/src/lib.rs:651]: this will cause a conflict with #5876 

```suggestion
    /// TODO(clabby): Implement this feature.
```

File: crates/storage/db/src/tables/codecs/postcard.rs
Comment (mattsse) [inline crates/storage/db/src/tables/codecs/postcard.rs:1]: why is this removed?

**Review by mattsse — APPROVED** [review]
ty.

this does multiple things at once, I'd appreciate if we could break these kinds of prs up into multiple

File: crates/storage/db/src/tables/codecs/postcard.rs
Comment (DaniPopes) [reply crates/storage/db/src/tables/codecs/postcard.rs:1]: Completely unused
